tony-core/src/main/java/com/linkedin/tony/HistoryFileUtils.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.tony;
+
+import java.io.IOException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+
+
+public class HistoryFileUtils {
+  private static final Log LOG = LogFactory.getLog(HistoryFileUtils.class);
+
+  public static TonyJobMetadata createMetadataObj(Configuration yarnConf, String appId, long started, long completed,
+      boolean status) {
+    String jobStatus = status ? "SUCCEEDED" : "FAILED";
+    String url = "http://" + yarnConf.get(YarnConfiguration.RM_WEBAPP_ADDRESS) + "/cluster/app/" + appId;
```

Comment:
make this a method
```
buildRMUrl(appId) {
return "http://" + yarnConf.get(YarnConfiguration.RM_WEBAPP_ADDRESS) + "/cluster/app/" + appId;
}
```

Reply:
If that's the case then I think this belongs in the generic `Utils` since it's not very specific to generating history file and might be useful for TonyClient. 
```Java
private void logTrackingAndRMUrls(ApplicationReport report) {
    LOG.info("URL to track running application (will proxy to TensorBoard once it has started): "
             + report.getTrackingUrl());
    LOG.info("ResourceManager web address for application: "
        + String.format(RM_APP_URL_TEMPLATE,
        yarnConf.get(YarnConfiguration.RM_WEBAPP_ADDRESS),
        report.getApplicationId())); // refactor this line to use `buildRMUrl(appId)`
  }
```

Reply:
Yah, looks better to be in Utils class